### PR TITLE
feat(replays): lower max replay recording segment size to 10 MiB

### PR DESCRIPTION
**Pull request by JoshFerge**
- [x] Update the config to include two separate values for maximum replay size: one for compressed size (called `max_replay_compressed`) and one for uncompressed size `max_replay_uncompressed_size`. The original value now represents the uncompressed size (`100 MiB`), which is utilized during processing. The new value, `max_replay_compressed`, is applied during the ingest stage.
- [x] Replays formerly relied on the maximum attachment limit of 100 MiB. Now, they use the new max_replay_compressed_size (`10 MiB`) setting for check_envelope_size_limits.

We are making this change as we have done analysis around our payload sizes, and if replays are above this size there is usually something off with them (mass mutation bugs, accidental inclusion of files, old SDK versions, etc.)

### CODE REVIEW

**Review by bruno-garcia — APPROVED** [review]
RSLGTM!

File: CHANGELOG.md
Comment (olksdr) [inline CHANGELOG.md]: It doesn't seem like a breaking change to me.

You just changing the defaults and the configuration itself will be done somewhere else, which can still contain the old numbers.  And also can be rolled out completely unrelated to this PR.  From the Relay POV nothing breaking is happening.

File: relay-config/src/config.rs
Comment (olksdr) [inline relay-config/src/config.rs:546]: Can we keep old config  and use `max_replay_size` as incoming check, the one is currently called `max_replay_compressed_size`, which clearly signals what the check is for, and if this is really necessary also use `serde` alias to rename it to the new name?

What do you think?

File: relay-config/src/config.rs
Comment (olksdr) [inline relay-config/src/config.rs]:
```suggestion
    /// Returns the maximum payload size for a compressed replay.
```

**Review by olksdr — CHANGES_REQUESTED** [review]
I left few questions, before approving this. 

Also you like to get @jjbayer opinion on the naming here.

File: relay-config/src/config.rs
Comment (jjbayer) [reply relay-config/src/config.rs:546]: The new naming looks OK to me, but I would add the old name as an `alias` as @olksdr suggests, such that this is not a breaking change.

File: CHANGELOG.md
Comment (jjbayer) [reply CHANGELOG.md]: +1 on moving this line to `** Internal **`.

File: CHANGELOG.md
Comment (jjbayer) [inline CHANGELOG.md]:
```suggestion
- Lower default max compressed replay recording segment size to 10 MiB. ([#2031](https://github.com/getsentry/relay/pull/2031))
```

File: CHANGELOG.md
Comment (olksdr) [inline CHANGELOG.md]: Please, merge master in and move this record to `Unreleased` section.

**Review by olksdr — COMMENTED** [review]
Once the changelog updated , this should be good to go.

**Review by jjbayer — APPROVED** [review]
See @olksdr 's changelog comments, otherwise LGTM!

**Comment by JoshFerge** [comment]
@bmckerry has increased the max message size of our kafka topic, this and https://github.com/getsentry/relay/pull/2032 should be good to go! can merge once approved. can deploy this one first, then 2032 subsequently.

**Review by olksdr — APPROVED** [review]

**Pull request merged**
